src/project/ithumbnailcreator.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include "modularity/imoduleinterface.h"
+#include "global/async/channel.h"
+#include "global/types/ret.h"
+#include "global/io/path.h"
+
+namespace au::project {
+
+class IThumbnailCreator : MODULE_EXPORT_INTERFACE
+{
+    INTERFACE_ID(au::project::ithumbnailcreator)
+
+public:
+    virtual ~IThumbnailCreator() = default;
+    virtual void onThumbnailCreated(bool success) = 0;
+    virtual muse::Ret createThumbnail(const muse::io::path_t& path) = 0;
+    virtual muse::async::Channel<muse::io::path_t> captureThumbnailRequested() const = 0;
```

Comment:
here are the details of the implementation revealed... I think it's not worth complicating it and solving it, I just suggest adding a comment and regrouping it somehow like this 

```
    virtual muse::Ret createThumbnail(const muse::io::path_t& path) = 0;

    // for internal use 
    virtual muse::async::Channel<muse::io::path_t> captureThumbnailRequested() const = 0;
    virtual void onThumbnailCreated(bool success) = 0;
```

Reply:
Done.

Reply:
mistake in grouping :)

Reply:
Done!

Reply:
Sorry for that.